paddle/majel/allocation.cc

```diff
@@ -0,0 +1,97 @@
+#include <boost/variant.hpp>
+
+#include "allocation.h"
+#include "hl_cuda.h"
+#include "paddle/utils/Logging.h"
+
+namespace majel {
+namespace detail {
+
+class Allocator : public boost::static_visitor<void*> {
+public:
+  Allocator(size_t size) : size_(size) {}
+
+  void* operator()(const CpuPlace& p) const {
+    void* address;
+    CHECK_EQ(posix_memalign(&address, 32ul, size_), 0);
+    CHECK(address) << "Fail to allocate CPU memory: size=" << size_;
+    return address;
+  }
+
+  void* operator()(const GpuPlace& p) const {
+    void* address = hl_malloc_device(size_);
```

Comment:
I think we can move the definition of [`hl_maclloc_device`](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/cuda/src/hl_cuda_device.cc#L269) to `paddle/majel/malloc.cc`.  The definition has only 4 lines of code.  In this way, Majel doesn't rely on `paddle/cuda`.